community/tutorials/gdnative/gdnative-c-example.rst

```diff
@@ -0,0 +1,338 @@
+.. _doc_gdnative_c_example:
+
+GDNative C example
+==================
+
+Introduction
+------------
+This tutorial will introduce you to the bare minimum required to create GDNative modules. This should be your starting point into the world of GDNative, understanding the contents of this tutorial will help you in understanding all that is to come after this.
+
+Before we begin, you can download the source code to the example object we'll be describing here by following this link:
+https://github.com/GodotNativeTools/GDNative-demos/tree/master/c/SimpleDemo
+
+This example project also contains a SConstruct file that makes compiling a little easier but in this tutorial we'll be doing things by hand.
+
+GDNative can be used to create several types of additions to Godot, from PluginScript to ARVR interfaces. In this tutorial we are going to look at creating a NativeScript module. NativeScript allows you to write logic in C or C++ in similar fashion as you would write a GDScript file. We'll be creating the C equivalent of this GDScript:
```

Comment:
You could once link to the (currently useless) API docs:

This tutorial will introduce you to the bare minimum required to create :ref:`GDNative<class_GDNative>` modules. This should be your starting point into the world of GDNative, understanding the contents of this tutorial will help you in understanding all that is to come after this.

Before we begin, you can download the source code to the example object we'll be describing here by following this link:
https://github.com/GodotNativeTools/GDNative-demos/tree/master/c/SimpleDemo

This example project also contains a SConstruct file that makes compiling a little easier but in this tutorial we'll be doing things by hand.

GDNative can be used to create several types of additions to Godot, from PluginScript to ARVR interfaces. In this tutorial we are going to look at creating a :ref:`NativeScript<class_NativeScript>` module. NativeScript allows you to write logic in C or C++ in similar fashion as you would write a GDScript file. We'll be creating the C equivalent of this GDScript: